src/control/system/database_pools.go

```diff
@@ -43,13 +46,25 @@ type (
 	// PoolServiceState is used to represent the state of the pool service
 	PoolServiceState uint
 
+	// PoolServiceStorage holds information about the pool storage.
+	PoolServiceStorage struct {
+		sync.Mutex
+		CreationRankStr string   // string rankset set at creation
+		creationRanks   *RankSet // used to reconstitute the rankset
+		CurrentRankStr  string   // string rankset representing current ranks
+		currentRanks    *RankSet // used to reconstitute the rankset
```

Comment:
The opposite of what I'd expect, as far as public/private. I'm curious why the strings are saved at all. Can't we generate them from the RankSets?

Reply:
the canonical form of a rank set is its ranged string so probably isn't necessary to store the rich type IMO

Reply:
It's more space efficient to store the string form. Also much easier to serialize for snapshots. The public/private here isn't really for external users, it's to control which fields are exposed to json. I guess I could use tags to make this more explicit.